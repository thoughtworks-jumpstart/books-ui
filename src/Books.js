import React from "react";

const Books = ({ data = [] }) => {
  return (
    <div>
      <h1>Books</h1>
      {data.map(book => {
        return <li key={book._id}>{book.title}</li>;
      })}
    </div>
  );
};

export default Books;
