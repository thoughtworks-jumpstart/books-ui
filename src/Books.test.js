import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import Books from "./Books.js";

it("renders a list of books", () => {
  const books = [
    {
      title: "Harry Potter",
      _id: "123"
    },
    {
      title: "Game of Thrones",
      _id: "456"
    }
  ];

  const renderer = new ShallowRenderer();
  renderer.render(<Books data={books} />);
  const output = renderer.getRenderOutput();

  expect(output).toMatchSnapshot();
});
