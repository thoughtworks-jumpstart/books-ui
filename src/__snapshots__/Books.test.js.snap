// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of books 1`] = `
<div>
  <h1>
    Books
  </h1>
  <ul>
    <li>
      Harry Potter
    </li>
    <li>
      Game of Thrones
    </li>
  </ul>
</div>
`;
